pkg/controllers/machine/garbagecollection/controller.go

```diff
@@ -68,26 +68,30 @@ func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 	cloudProviderProviderIDs := sets.New[string](lo.Map(cloudProviderMachines, func(m *v1alpha5.Machine, _ int) string {
 		return m.Status.ProviderID
 	})...)
-	machines := lo.Filter(lo.ToSlicePtr(machineList.Items), func(m *v1alpha5.Machine, _ int) bool {
-		return m.StatusConditions().GetCondition(v1alpha5.MachineLaunched).IsTrue() &&
-			m.DeletionTimestamp.IsZero() &&
-			c.clock.Since(m.StatusConditions().GetCondition(v1alpha5.MachineLaunched).LastTransitionTime.Inner.Time) > time.Second*10 &&
-			!cloudProviderProviderIDs.Has(m.Status.ProviderID)
+	nodeClaims := lo.Filter(lo.ToSlicePtr(nodeClaimList.Items), func(n *v1beta1.NodeClaim, _ int) bool {
+		return n.StatusConditions().GetCondition(v1beta1.NodeLaunched).IsTrue() &&
+			n.DeletionTimestamp.IsZero() &&
+			c.clock.Since(n.StatusConditions().GetCondition(v1beta1.NodeLaunched).LastTransitionTime.Inner.Time) > time.Second*10 &&
+			!cloudProviderProviderIDs.Has(n.Status.ProviderID)
 	})
 
-	errs := make([]error, len(machines))
-	workqueue.ParallelizeUntil(ctx, 20, len(machines), func(i int) {
-		if err := c.kubeClient.Delete(ctx, machines[i]); err != nil {
+	errs := make([]error, len(nodeClaims))
+	workqueue.ParallelizeUntil(ctx, 20, len(nodeClaims), func(i int) {
+		if err := nodeclaimutil.Delete(ctx, c.kubeClient, nodeClaims[i]); err != nil {
 			errs[i] = client.IgnoreNotFound(err)
 			return
 		}
 		logging.FromContext(ctx).
-			With("provisioner", machines[i].Labels[v1alpha5.ProvisionerNameLabelKey], "machine", machines[i].Name, "provider-id", machines[i].Status.ProviderID).
-			Debugf("garbage collecting machine with no cloudprovider representation")
-		metrics.MachinesTerminatedCounter.With(prometheus.Labels{
-			metrics.ReasonLabel:      "garbage_collected",
-			metrics.ProvisionerLabel: machines[i].Labels[v1alpha5.ProvisionerNameLabelKey],
-		}).Inc()
+			With(
+				lo.Ternary(nodeClaims[i].IsMachine, "machine", "nodeclaim"),
+				nodeClaims[i].Name,
+				"provider-id",
+				nodeClaims[i].Status.ProviderID,
+				lo.Ternary(nodeClaims[i].IsMachine, "provisioner", "nodepool"),
+				nodeclaimutil.OwnerName(nodeClaims[i]),
```

Comment:
nit: personally think it'd be more readable to have this in 3 lines for the 3 context values, but up to you.
```suggestion
				lo.Ternary(nodeClaims[i].IsMachine, "machine", "nodeclaim"), nodeClaims[i].Name,
				"provider-id", nodeClaims[i].Status.ProviderID,
				lo.Ternary(nodeClaims[i].IsMachine, "provisioner", "nodepool"), nodeclaimutil.OwnerName(nodeClaims[i]),
```

Reply:
Agreed, good point.